### PR TITLE
Cordic api modify

### PR DESCRIPTION
**Pull request by behappy1234**
## Summary
small changes below
1. add scale in cordic api, because some hardware accelerator has scale parameter in cordic funtions. 
2. remove CONFIG_MATH_CORDIC_USE_Q31 for cordic_num_t, it's redundant, especially if you open CONFIG_MATH_CORDIC and not open for CONFIG_MATH_CORDIC_USE_Q31, the compile will fail.

## Impact
include/nuttx/math

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
> CONFIG_MATH_CORDIC_USE_Q31 is not redundant. It's CORDIC implementation feature and your lower-half implementation should select this option if needed. There are CORDIC peripherals that support q1.15 (int16_t) and this option allows you to implement other numerical types.

@raiden00pl but if defconfig doesn't enable CONFIG_MATH_CORDIC_USE_Q31, the compiler will complain that cordic_num_t at https://github.com/apache/nuttx/pull/10282/files#diff-2935a1615b202c6a7277770f43b933c03916d03bfc303ec732b12dcadbf9737fR126 is undefined.

**Comment by raiden00pl** [comment]
If your driver use Q31 for CORDIC then why not  `select MATH_CORDIC_USE_Q31` in your lower-half CORDIC implementation ? If your driver return data in other format then you have to implement `CONFIG_MATH_CORDIC_USE_{any_other_float_format}`.

**Comment by raiden00pl** [comment]
BTW not defining CORDIC data type has no sense. You can't write a portable application if you don't know CORDIC data type.

File: include/nuttx/math/cordic.h [hunk 1]
```diff
@@ -119,6 +117,8 @@ struct cordic_calc_s
   /* CORDIC request configuration */
 
   uint8_t func;                 /* CORDIC function */
+  int8_t  input_scale;
```
[the remark this answers is not in the document]
Comment (behappy1234) [reply include/nuttx/math/cordic.h:120]: I will add "cordic scale factor".

**Comment by behappy1234** [comment]
> If your driver use Q31 for CORDIC then why not `select MATH_CORDIC_USE_Q31` in your lower-half CORDIC implementation ? If your driver return data in other format then you have to implement `CONFIG_MATH_CORDIC_USE_{any_other_fractionalnum_format}`.

**Comment by behappy1234** [comment]
> CONFIG_MATH_CORDIC_USE_Q31 is not redundant. It's CORDIC implementation feature and your lower-half implementation should select this option if needed. There are CORDIC peripherals that support q1.15 (int16_t) and this option allows you to implement other numerical types.

Yes, the numerical type for q1.15 is as a default here. I do this, because if I open for MATH_FFT, then math.c is compiled, so cordic.h is included and CONFIG_MATH_CORDIC_USE_Q31 is needed for cordic_num_t to be configed even in this case I don't need to care about cordic. So, CONFIG_MATH_CORDIC_USE_Q31 is "redundant" here, or a default cordic_num_t definition is need. And I remove it as a default one.

**Pull request closed without merging**